logbus/solvermon/vertexmon.go

```diff
@@ -73,6 +74,10 @@ func (vm *vertexMonitor) parseError() {
 		internalStr = " internal"
 	}
 	switch {
+	case reHint.MatchString(errString):
+		matches, _ := stringutil.NamedGroupMatches(errString, reHint)
+		errString = matches["msg"][0]
```

Comment:
@alexcb , @mikejholly 
Any idea why the linter is throwing a warning for this line:
```
logbus/solvermon/vertexmon.go:79:3: ineffectual assignment to errString (ineffassign)
```

Reply:
It's probably not smart enough to detect the `fallthrough` below. I think these can be ignored using a special comment.

Reply:
is `errString` being used anywhere after? e.g.

```
func() int {
  x := 1
  y := x * 2
  x = 10  /// this would be an example of a meaningless assignment
  return y
}

Reply:
right, but it seems clear that it is (unless I'm missing something)

Reply:
added a comment to ignore

Reply:
I think this illustrates what's going on:

```
func main() {
	s := "a test"
	switch {
	case strings.Contains(s, "a"):
		s = "greetings"
		fallthrough
	case strings.Contains(s, "o"):
		fmt.Printf("this was hit; %s\n", s)
	}
}
```

this will still print `this was hit; greetings` since the `fallthrough` skips evaluating `strings.Contains(s, "o")`, but rather jumps to the block of the switch and executes the Printf.


in your specific case it turns out to be executing:

```
		matches, _ := stringutil.NamedGroupMatches(errString, reHint)
		errString = matches["msg"][0] //nolint:ineffassign,staticcheck
		vm.isCanceled = true
		vm.errorStr = "WARN: Canceled"
		return
```

which doesn't make use of `errString` after it's assigned to.

I think you should be able to delete this line rather than disable the linter.

Reply:
Oh, that's interesting. I didn't know that `fallthrough` skips conditionals. 

https://go.dev/play/p/e1lldiHKKP2

Reply:
yeah.... this could very well be a bug if @idodod is expecting the rest of the switch cases to be evaluated here.

If that is the case, then this logic should be done before the switch statement.

Reply:
yeah, exactly, I was expecting for the next condition to be checked. Thanks for clarifying. 

> If that is the case, then this logic should be done before the switch statement.

I will do just that. Thanks!

Reply:
ok, fixed! I learned something today!